js/apps/system/_admin/aardvark/APP/frontend/js/arango/arangoValidationHelper.js

```diff
@@ -0,0 +1,143 @@
+/* jshint unused: false */
+(function () {
+  "use strict";
+
+  window.arangoValidationHelper = {
+    getControlCharactersRegex: () => {
+      return /[^\u0000-\u001F]/;
+    },
+    getDocumentKeySpecialCharactersValidation: () => {
+      var keySpecialCharactersValidation = {
+        rule: Joi.string().regex(/^[a-zA-Z0-9_\-:\.@()\+,=;$!*\%']+$/),
```

Comment:
nit: percent character does not need to be escaped
```suggestion
        rule: Joi.string().regex(/^[a-zA-Z0-9_\-:\.@()\+,=;$!*%']+$/),
```